assertj-core/src/main/java/org/assertj/core/util/BiComparator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2023 the original author or authors.
+ */
+package org.assertj.core.util;
+
+/**
+ * A comparison function among two different types.
+ *
+ * @param <T> the type of first objects that may be compared by this comparator
+ * @param <U> the type of second objects that may be compared by this comparator
+ * @author Alessandro Modolo
+ */
+@FunctionalInterface
+public interface BiComparator<T, U> {
```

Comment:
Maybe not needed if we focus only on `BiPredicate`

Reply:
removed in 122d8af03bf46ff5716bf578d37b0cf6da4bf4aa